CountlyTests/CountlyViewTests.swift

```diff
@@ -10,190 +10,506 @@ import XCTest
         let config = createBaseConfig()
-        // No Device ID provided during init
         Countly.sharedInstance().start(with: config)
         
-        let viewID = Countly.sharedInstance().views().startAutoStoppedView("TestAutoStoppedView")
+        let viewID = Countly.sharedInstance().views().startView("View1", segmentation: ["key": "value"])
+        XCTAssertNotNil(viewID, "View should be started successfully with segmentation.")
         
-        let pauseViewExpectation = XCTestExpectation(description: "Wait for pause view")
-        DispatchQueue.global().asyncAfter(deadline: .now() + 5) {
-            Countly.sharedInstance().views().pauseView(withID: viewID)
-            pauseViewExpectation.fulfill()
+        let expectation = XCTestExpectation(description: "Wait for 4 seconds before stopping the view.")
+        DispatchQueue.main.asyncAfter(deadline: .now() + 4) {
+            Countly.sharedInstance().views().stopView(withName: "View1")
+            expectation.fulfill()
         }
         
-        let resumeViewExpectation = XCTestExpectation(description: "Wait for resume view")
-        DispatchQueue.global().asyncAfter(deadline: .now() + 10) { // Delayed by 10 seconds
-            Countly.sharedInstance().views().resumeView(withID: viewID)
-            resumeViewExpectation.fulfill()
-        }
+        wait(for: [expectation], timeout: 5.0) // Wait for the expectation to be fulfilled
+        checkRecordedEventsForView(viewName: "View1", segmentation: ["key": "value"])
+    }
+    
+    func testStartViewAndStopViewWithID() throws {
+        let config = createBaseConfig()
+        Countly.sharedInstance().start(with: config)
         
-        let bgExpectation = XCTestExpectation(description: "Wait for background notification")
-        DispatchQueue.global().asyncAfter(deadline: .now() + 15) { // Delayed by 15 seconds
-            NotificationCenter.default.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
-            bgExpectation.fulfill()
+        let viewID = Countly.sharedInstance().views().startView("View1") ?? ""
+        XCTAssertNotNil(viewID, "View should be started successfully.")
+        
+        let expectation = XCTestExpectation(description: "Wait for 3 seconds before stopping the view.")
+        DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
+            Countly.sharedInstance().views().stopView(withID: viewID)
+            expectation.fulfill()
         }
         
-        let fgExpectation = XCTestExpectation(description: "Wait for active notification")
-        DispatchQueue.global().asyncAfter(deadline: .now() + 20) { // Delayed by 20 seconds
-            NotificationCenter.default.post(name: UIApplication.didBecomeActiveNotification, object: nil)
-            fgExpectation.fulfill()
+        wait(for: [expectation], timeout: 5.0) // Wait for the expectation to be fulfilled
+        checkRecordedEventsForView(withID: viewID)
```

Comment:
we want to see specifically also those durations are validated and SDK correctly calculates the duration